### PR TITLE
Add code to ignore OSC commands

### PR DESCRIPTION
**Pull request by tannewt**
CircuitPython 8 uses OSC to set title.

### CODE REVIEW

**Comment by carlosperate** [comment]
Thanks for the PR @tannewt! 

PR https://github.com/mu-editor/mu/pull/2323 has been merged, so rebasing or merging main should resolve the CI failures.

I'm not that familiar with OSC commands, is this a generic way to change a terminal title bar?

File: tests/interface/test_panes.py [hunk 1]
```diff
@@ -898,6 +898,73 @@ def test_MicroPythonREPLPane_process_tty_data_unsupported_vt100_command():
...
+    rp.process_tty_data(bs)
+    assert rp.unprocessed_input == ""
+    assert rp.toPlainText() == "Hello world!"
+    assert rp.textCursor().position() == 5
+    assert rp.device_cursor_position == 5
+
+
+def test_MicroPythonREPLPane_process_tty_data_osc_title_command():
+    """
+    Ensure nothing is done, when receiving an unsupported OSC command
```
Comment (carlosperate) [inline tests/interface/test_panes.py:930]: Minor nitpick but this comment and the one in line 938 probably need to be changed for this unit test.

**Comment by tannewt** [comment]
> Thanks for the PR @tannewt!

np, thanks for the fast review!

> PR #2323 has been merged, so rebasing or merging main should resolve the CI failures.

Done! 

> I'm not that familiar with OSC commands, is this a generic way to change a terminal title bar?

Correct OSC commands are Operating System Commands that were defined by xterm early on. A couple of them are specific to setting the terminal title. (Icon name is the text for the window's icon in a task bar.) Here are two good references:
* [xterm control sequences](https://www.xfree86.org/current/ctlseqs.html)
* [terminal guide](https://terminalguide.namepad.de/seq/)

File: mu/interface/panes.py [hunk 2]
```diff
@@ -182,6 +182,9 @@ def __init__(self, connection, theme="day", parent=None):
         self.vt100_regex = re.compile(
             r"\x1B\[(?P<count>[\d]*)(;?[\d]*)*(?P<action>[A-Za-z])"
         )
+        self.osc_regex = re.compile(
+            r"\x1B\](?P<command>[\d]*);(?P<string>[^\x1B]*)\x1B\\"
```
Comment (carlosperate) [inline mu/interface/panes.py:186]: Should this regex not start with `r"\x1B\[`?

Comment (carlosperate) [reply mu/interface/panes.py:186]: No wait, I was looking at the wrong command, nevermind.

**Review by carlosperate — APPROVED** [review]
Looks good! 👍
Tested with a real device as well and works well, thanks!

**Pull request merged**